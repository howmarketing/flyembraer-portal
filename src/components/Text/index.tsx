import { ReactNode } from "react"
import { styled } from "../../../stitches.config"

export type TextProps = {
	as?: React.ElementType
	children: ReactNode
	color?:
		| "black"
		| "gray100"
		| "gray200"
		| "gray300"
		| "white"
		| "brand-primary"
		| "brand-secondary"
		| "error"
		| "success"
	size?: "xsm" | "sm" | "md" | "lg" | "xl" | "xxl"
	weight?: "thin" | "normal" | "medium" | "bold"
} & React.ComponentProps<typeof StyledText>

const Text = ({ children, color, size, weight, ...props }: TextProps) => {
	return (
		<StyledText color={color} size={size} weight={weight} {...props}>
			{children}
		</StyledText>
	)
}

export default Text

const StyledText = styled("span", {
	lineHeight: "$1",

	variants: {
		color: {
			white: {
				color: "$white",
			},
			black: {
				color: "$black200",
			},
			gray100: {
				color: "$gray100",
			},
			gray200: {
				color: "$gray200",
			},
			gray300: {
				color: "$gray300",
			},
			"brand-primary": {
				color: "$brand-primary",
			},
			"brand-secondary": {
				color: "$brand-secondary",
			},
			error: {
				color: "$error",
			},
			success: {
				color: "$success",
			},
		},
		size: {
			xsm: {
				fontSize: "$xsm",
			},
			sm: {
				fontSize: "$sm",
			},
			md: {
				fontSize: "$md",
			},
			lg: {
				fontSize: "$lg",
			},
			xl: {
				fontSize: "$xl",
				lineHeight: "$2",
			},
			xxl: {
				fontSize: "$xxl",
				lineHeight: "$3",
			},
		},
		weight: {
			thin: {
				fontWeight: "$thin",
			},
			normal: {
				fontWeight: "$normal",
			},
			medium: {
				fontWeight: "$medium",
			},
			bold: {
				fontWeight: "$bold",
			},
		},
	},
	defaultVariants: {
		color: "black",
		size: "sm",
		weight: "normal",
	},
})
