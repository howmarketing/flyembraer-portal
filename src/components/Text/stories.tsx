import { ComponentMeta, ComponentStory } from "@storybook/react"

import Text from "."

export default {
	title: "Components/Text",
	component: Text,
	args: {
		children: "Label",
	},
	argTypes: {
		size: {
			defaultValue: "md",
			options: ["xsm", "sm", "md", "lg", "xl"],
			control: { type: "select" },
		},
		color: {
			defaultValue: "black",
			options: ["black", "gray100", "gray200"],
			control: { type: "inline-radio" },
		},
		weight: {
			defaultValue: "normal",
			options: ["normal", "bold"],
			control: { type: "inline-radio" },
		},
	},
} as ComponentMeta<typeof Text>

const Template: ComponentStory<typeof Text> = (args) => <Text {...args}>{args.children}</Text>

export const Default = Template.bind({})

export const As_Element = Template.bind({})
As_Element.args = {
	as: "p",
}
