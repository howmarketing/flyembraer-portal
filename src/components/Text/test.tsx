import { render, screen } from "@testing-library/react"

import Text from "."

describe("<Text />", () => {
	it("should render the Text and match the previous version", () => {
		const { container } = render(<Text>Label</Text>)

		expect(screen.getByText(/Label/i)).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it("should render the text as another element", () => {
		render(<Text as="h1">Label</Text>)

		expect(screen.getByRole("heading", { name: /Label/i })).toBeInTheDocument()
	})
})
