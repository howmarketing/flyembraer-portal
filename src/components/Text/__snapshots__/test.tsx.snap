// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render the Text and match the previous version 1`] = `
<span
  class="c-kbxdZX c-kbxdZX-cHtIMp-color-black c-kbxdZX-fAKXXr-size-sm c-kbxdZX-gVAYaD-weight-normal"
>
  Label
</span>
`;
